src/coreclr/src/pal/src/loader/module.cpp

```diff
@@ -592,6 +604,11 @@ PALAPI
 PAL_LoadLibraryDirect(
     IN LPCWSTR lpLibFileName)
 {
+    if (ShouldRedirectToCurrentLibrary(lpLibFileName))
+    {
```

Comment:
Nit: This should be after the `ENTRY` macro to follow PAL conventions

Reply:
Good to know.

Reply:
Also should not return without `LOGEXIT` and `PERF_EXIT` . Fixed that too.